src/hitch.c

```diff
@@ -461,7 +461,7 @@ handle_shcupd(struct ev_loop *loop, ev_io *w, int revents)
 		/* drop too unsync updates */
 		r -= sizeof(uint32_t);
 		encdate = *((uint32_t *)&msg[r]);
-		if (!(abs((int)(int32_t)now - ntohl(encdate))
+		if (!((int)(int32_t)now - ntohl(encdate)
```

Comment:
I've always wondered about the purpose of this double casting.  I think a couple of parenthesis were dropped along the way.  The actual code should really be:
`abs((int)((int32_t)now - ntohl(encdate))`

Suppose that the peer receiving the update has a clock drifting by 1s compared to the one sending it.  To simplify, imagine that `now ==ntohl(encdate) - 1`.  With the current fix, the update would just be discarded.  Of course, I might be missing something really obvious.